### PR TITLE
Improve TTL handling on screen

### PR DESCRIPTION
**Pull request by leongb**

### CODE REVIEW

File: app/src/parser.c [hunk 1]
```diff
@@ -489,22 +489,61 @@ parser_error_t parser_getItem(parser_context_t *ctx,
...
-                    uint64_t days = hours/24;
-                    fpuint64_to_str(tmpBuffer, sizeof(tmpBuffer), days, 0);
-                    snprintf(outVal, outValLen, "%sday", tmpBuffer);
-                    return parser_ok;
+            value /= 1000;
+            char buffer[100];
+            MEMZERO(buffer,sizeof(buffer));
+            uint16_t index = 0;
+            uint64_t days = value / (60*60*24);
+            if(days > 28){
```
Comment (jleni) [inline app/src/parser.c:497]: Is the limit 29 or 28?
Example, is 28days 23 hours valid?

File: app/src/parser.c [hunk 2]
```diff
@@ -489,22 +489,61 @@ parser_error_t parser_getItem(parser_context_t *ctx,
             CHECK_PARSER_ERR(index_headerpart(parser_tx_obj.header, header_ttl, &ctx->offset));
             uint64_t value = 0;
             CHECK_PARSER_ERR(readU64(ctx,&value));
```
Comment (jleni) [inline app/src/parser.c:491]: I think we should move all this code into a function that can be unit tested

**Review by jleni — CHANGES_REQUESTED** [review]
Let's confirm that 28d 23h is fine and let's refactor into a function

File: app/src/parser.c (hunk 1 above)
Comment (leongb) [reply app/src/parser.c]: checked with them and what we have is fine for now

**Pull request merged**